velox/docs/develop/testing/fuzzer.rst

```diff
@@ -1,64 +1,106 @@
-=================
-Expression Fuzzer
-=================
-
-Velox allows users to define UDFs and provides an expression fuzzer tool to
-test expression evaluation engine and UDFs thoroughly. It is being used to
-test builtin Presto and Spark UDFs and has discovered numerous bugs caused
-by corner cases that are difficult to cover in unit tests.
-
-The expression fuzzer generates random expressions and evaluates these on
-random input vectors. Each generated expression may contain multiple sub-
-expressions and each input vector can have random and potentially nested
-encodings.
-
-To ensure that evaluation engine and UDFs handle vector encodings correctly,
-the expression fuzzer evaluates an expression twice and asserts the results to
-be the same: using regular evaluation path and using simplified evaluation that
+======
+Fuzzer
+======
+
+Velox allows users to define UDFs (user-defined functions) and UDAFs
+(user-defined aggregate functions) and provides a fuzzer tools to test the
+engine and UDFs thoroughly. These tools are being used to test builtin Presto
+and Spark functions and have discovered numerous bugs caused by corner cases
+that are difficult to cover in unit tests.
+
+The Expression Fuzzer tests the expression evaluation engine and UDFs by
+generating random expressions and evaluating these on random input vectors.
+Each generated expression may contain multiple sub-expressions and each input
+vector can have random and potentially nested encodings.
+
+To ensure that evaluation engine and UDFs handle vector encodings correctly, the
+expression fuzzer evaluates each expression twice and asserts the results to be
+the same: using regular evaluation path and using simplified evaluation that
 flattens all input vectors before evaluating an expression.
 
-How to integrate with expression fuzzer
+The Aggregation Fuzzer tests the HashAggregation operator and UDAFs by
+generating random aggregations and evaluating these on random input vectors.
+
+The Aggregation Fuzzer tests global aggregations (no grouping keys), group-by
+aggregations (one or more grouping keys), distinct aggregations(no aggregates),
+aggregations with and without masks.
+
+The results of aggregations using functions supported by DuckDB are compared
+with DuckDB results.
+
+For each aggregation, Fuzzer generates multiple logically equivalent plans and
+verifies that results match. These plans are:
+
+- Single aggregation (raw input, final result).
```

Comment:
Do we plan to cover spilling next? Thanks!

Reply:
That would be nice: #3204